clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/RequestBasedMetadata.java

```diff
@@ -155,15 +157,14 @@ private synchronized boolean updateCache(boolean onDemandRefresh) throws Interru
     long currentTimeMs = System.currentTimeMillis();
     // call the METADATA endpoint
     try {
-      byte[] body = fetchMetadata().get().getBody();
+      byte[] body = fetchMetadata(mostRecentHash).get().getBody();
       RecordDeserializer<MetadataResponseRecord> metadataResponseDeserializer = FastSerializerDeserializerFactory
           .getFastAvroSpecificDeserializer(MetadataResponseRecord.SCHEMA$, MetadataResponseRecord.class);
       MetadataResponseRecord metadataResponse = metadataResponseDeserializer.deserialize(body);
       VersionProperties versionMetadata = metadataResponse.getVersionMetadata();
 
-      int fetchedVersion = versionMetadata.getCurrentVersion();
-
-      if (fetchedVersion != getCurrentStoreVersion()) {
```

Comment:
can remove the version comparison because it will be null if the versions were supposed to be unchanged